### PR TITLE
[CBO-192] Provisional assessment MI report

### PR DESCRIPTION
**Pull request by lostie**
#### Ticket

[Provisional Assessment MI report](https://dsdmoj.atlassian.net/browse/CBO-192)

#### Improvements for the future

 - [ ] [Persist reports in S3 rather than as they are currently (database blob)](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/issues/2431)

### CODE REVIEW

**Comment by lostie** [comment]
Currently the new Management Information page looks like this:

screen shot 2018-07-18 at 11 09 22

File: spec/models/stats/management_information_generator_spec.rb [hunk 1]
```diff
@@ -1,61 +1,22 @@
...
+  context 'data generation' do
+    let!(:valid_claims) {
+      [
+        create(:allocated_claim),
+        create(:authorised_claim),
+        create(:part_authorised_claim)
+      ]
+    }
+    let!(:draft_claim) { create(:draft_claim) }
+    let!(:non_active_claim) { Timecop.freeze(frozen_time) { create(:allocated_claim) } }
```
Comment (jsugarman) [inline spec/models/stats/management_information_generator_spec.rb:15]: 💭 can use the rails equivalent for time travel instead (we have been looking to switch timecop to these) http://api.rubyonrails.org/v5.1/classes/ActiveSupport/Testing/TimeHelpers.html

File: spec/models/stats/stats_report_spec.rb [hunk 2]
```diff
@@ -1,136 +1,131 @@
...
+      it 'updates with report contents and completed_at time' do
+        frozen_time = Time.new(2015, 3, 16, 13, 36, 12)
+        record = nil
+        Timecop.freeze(frozen_time) do
+          record = described_class.record_start('my_new_report')
+        end
 
-          StatsReport.destroy_unfinished_reports_older_than('management_information', 2.hours.ago)
-          expect(StatsReport.all).to match_array [my_report_new, other_report_old, other_report_new]
+        Timecop.freeze(frozen_time + 2.minutes) do
```
Comment (jsugarman) [inline spec/models/stats/stats_report_spec.rb:94]: ditto on timecop

File: spec/models/stats/stats_report_spec.rb [hunk 3]
```diff
@@ -1,136 +1,131 @@
...
-        it 'destroys incomplete reports started before the timestamp' do
-          _my_report_old = create :stats_report, :incomplete, started_at: 121.minutes.ago
-          my_report_new = create :stats_report, :incomplete, started_at: 119.minutes.ago
-          other_report_old = create :stats_report, :other_report, :incomplete, started_at: 121.minutes.ago
-          other_report_new = create :stats_report, :other_report, :incomplete, started_at: 119.minutes.ago
+    describe '#write_report' do
+      it 'updates with report contents and completed_at time' do
+        frozen_time = Time.new(2015, 3, 16, 13, 36, 12)
+        record = nil
+        Timecop.freeze(frozen_time) do
```
Comment (jsugarman) [inline spec/models/stats/stats_report_spec.rb:90]: ditto on timecop

File: app/services/reports/provisional_assessments.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    def call
+      Stats::MIData.connection.execute(query).to_a
+    end
+
+    private
+
+    def query
+      %{SELECT supplier_data.provider_name, provider_type,
```
Comment (jsugarman) [inline app/services/reports/provisional_assessments.rb:18]: 💭 could use [squiggly heredoc](https://infinum.co/the-capsized-eight/multiline-strings-ruby-2-3-0-the-squiggly-heredoc) for better log format and readability but either way

**Review by jsugarman — APPROVED** [review]
👍 LGTM, couple of suggestions, no blockers

**Comment by lostie** [comment]
I'll try and replace the `Timecop` with Rails time travel helpers as part of other ticket that is also using it.

**Pull request merged**